### PR TITLE
feat - Migrate to Transformers 5.5.4

### PR DESCRIPTION
**Pull request by kappacommit**
## Summary

Migrates to transformers 5.5.4

I tried to migrate to the latest 5.9.0 but transformers introduced a breakage with 5.6.0 that breaks loading of SD1.5 checkpoints. I was not able to resolve this breakage. I created an issue against Diffusers here, I believe this needs resolved before we can progress past 5.5.4 https://github.com/huggingface/diffusers/issues/13833

## QA Instructions

Testing guide:

## P1 — Prompt weighting (Compel fork) — highest risk

**Changed:** `compel==2.1.1` (PyPI) → `invoke-ai/compel@main` fork (2.3.1), the
only compel build that supports transformers 5.x. This is the largest single
behavioural change in the migration.

| # | Test | Steps | Pass criteria |
|---|------|-------|---------------|
| W1 | SD 1.5 weighted prompt | SD 1.5 model. Prompt `a (red:1.5) car on a (blue:0.5) road`, 512×512, seed 42. | Generates without error; visibly more red than blue vs an unweighted run |
| W2 | SD 1.5 negative prompt | Prompt `a photo of a dog`, negative `blurry, low quality`. | Generates without crash |
| W3 | SDXL weighted prompt | SDXL model, same weighted prompt as W1. | Generates; weights applied |
| W4 | Long prompt (>77 tokens) | SD 1.5, a prompt well over 77 tokens. | Compel concatenates chunks; generates without tokenizer/length error |

---

## P2 — Z-Image text encoder (rope_theta fix) — new in this migration

**Changed:** `invokeai/backend/model_manager/load/model_loaders/z_image.py`.
transformers 5.x removed the top-level `Qwen3Config.rope_theta` attribute (the
value now lives in the `rope_parameters` dict). The loader reconstructs the
rotary-embedding `inv_freq` buffer from that base, so a wrong/missing value
silently corrupts positional encoding → garbled images. Covers both the
safetensors and GGUF Qwen3 encoder paths.

| # | Test | Steps | Pass criteria |
|---|------|-------|---------------|
| Z1 | Z-Image (safetensors) | Load a Z-Image model (safetensors text encoder). Prompt `a serene mountain lake at dawn`. | Generates a coherent image — no AttributeError, no noise/garbled output |
| Z2 | Z-Image (GGUF encoder) | Load a Z-Image model with a GGUF Qwen3 text encoder. Same prompt. | Generates a coherent image; console shows no "Re-initializing unknown meta buffer" warning for `inv_freq` |
| Z3 | Determinism check | Run Z1 twice with the same seed. | Identical output both runs (confirms rope base is stable, not defaulted) |

---

## P3 — HuggingFace token status (whoami) — new in this migration

**Changed:** `invokeai/app/api/routers/model_manager.py`. huggingface_hub 1.x
removed `get_token_permission()`; `HFTokenHelper.get_status()` now validates via
`whoami()` — VALID on success, INVALID on `HfHubHTTPError` (e.g. 401), UNKNOWN on
other errors (e.g. no network). CPU-side; runs on any machine.

- Remove your HF token
- Attempt to install a model via the model manager that is gated behind approval (eg: SD 3.5)
- Verify failure message tells you that you need a token
- Add an HF token
- Attempt to install the same model
- Verify success

---

## P4 — T5 / CLIP text encoders (tokenizer + weight-tying changes)

**Changed (5.1.0 migration, re-validated on 5.9):** `flux_text_encoder.py`,
`sd3_text_encoder.py`, T5 weight sharing now via `tie_weights()` rather than an
identity assertion.

| # | Test | Steps | Pass criteria |
|---|------|-------|---------------|
| T1 | FLUX text-to-image | FLUX schnell (fp8/GGUF for 8 GB). Prompt `a lighthouse on a cliff at sunset`. | Generates without AssertionError or tokenizer error |
| T2 | FLUX long prompt | Same model, prompt > 500 words. | Generates; truncation is a warning, not a crash |
| T3 | SD3 text-to-image | SD3 medium (if VRAM allows). Simple prompt. | Generates without tokenizer error |
| T4 | FLUX bnb-nf4 (quantized) | Load a FLUX bnb-nf4 checkpoint, generate. | No AssertionError in `_load_state_dict_into_t5()`; generates |

---

## P5 — Safety checker & CLIP image processor (diffusers pipeline)

**Changed (5.1.0 migration):** `safety_checker.py`, `diffusers_pipeline.py`
(feature-extractor → image-processor APIs).

| # | Test | Steps | Pass criteria |
|---|------|-------|---------------|
| S1 | Safety checker first download | Delete `<invoke_root>/models/core/convert/stable-diffusion-safety-checker/` if present. Enable NSFW checker. Generate SD 1.5 image. | Checker downloads; image generates (pass or blurred); no serialization error |
| S2 | Safety checker cached load | Generate another SD 1.5 image. | Loads from disk; no re-download |
| C1 | SD 1.5 baseline | Load SD 1.5, generate any image. | Generates without Import/AttributeError from the pipeline |

---

## Regression sweep

After the targeted tests, a broad smoke pass over the app:

- Generate on each model family you have installed (SD 1.5, SDXL, FLUX, SD3, Z-Image).
- Exercise one ControlNet and one IP-Adapter run if available (both import transformers vision models).
- Watch the console for any `ImportError`, `AttributeError`, or `AssertionError` — none expected.

## Checklist

- [x] _The PR has a short but descriptive title, suitable for a changelog_
- [x] _Tests added / updated (if applicable)_
- [ ] _❗Changes to a redux slice have a corresponding migration_
- [ ] _Documentation added / updated (if applicable)_
- [ ] _Updated `What's New` copy (if doing a release after this PR)_

### CODE REVIEW

File: uv.lock [hunk 1]
```diff
@@ -1233,6 +1496,218 @@ wheels = [
...
+    { name = "terminado" },
+]
+sdist = { url = "https://files.pythonhosted.org/packages/f4/a7/bcd0a9b0cbba88986fe944aaaf91bfda603e5a50bda8ed15123f381a3b2f/jupyter_server_terminals-0.5.4.tar.gz", hash = "sha256:bbda128ed41d0be9020349f9f1f2a4ab9952a73ed5f5ac9f1419794761fb87f5", size = 31770, upload-time = "2026-01-14T16:53:20.213Z" }
+wheels = [
+    { url = "https://files.pythonhosted.org/packages/d1/2d/6674563f71c6320841fc300911a55143925112a72a883e2ca71fba4c618d/jupyter_server_terminals-0.5.4-py3-none-any.whl", hash = "sha256:55be353fc74a80bc7f3b20e6be50a55a61cd525626f578dcb66a5708e2007d14", size = 13704, upload-time = "2026-01-14T16:53:18.738Z" },
+]
+
+[[package]]
+name = "jupyterlab"
```
Comment (kappacommit) [inline uv.lock:1559]: A good chunk of these dependencies are added due to our fork of compel adding the `notebook` dependency which pulls in the whole jupyter stack. 

We _probably_ don't need Jupyter on our compel fork. Or at most, only need it in dev and not as a core dependency. If we can get rid of that, the change in this file will trim down significantly (but not fully disappear.) 

The actual steps to resolve this would be:

#### In invoke-ai/compel
1. In pyproject.toml, remove notebook from the core dependencies. 
2. Commit + push to main. That produces a new commit SHA on the branch — nothing else needs to change, since Invoke pins rev=main.

#### On this branch after that commit lands
`uv lock --upgrade-package compel` and commit the updated uv lock

**Comment by kappacommit** [comment]
Updated to use official compel 2.4 instead of our fork. 

Also the issue I opened with diffusers looks like it is being addressed pretty quickly, we may be able to get to transformers 5.9.0 soon. https://github.com/huggingface/diffusers/pull/13843

**Review by Pfannkuchensack — APPROVED** [review]
Image

Tested with atleast one of each Model Type. Works fine.
The Huggingface changes needs some updates later.

**Review by lstein — APPROVED** [review]
